Assets/HoloToolkit/Input/Scripts/UI/Button.cs

```diff
@@ -0,0 +1,237 @@
+        {
+            get { return selected; }
+            set
+            {
+                if (selected != value)
+                {
+                    selected = value;
+                    UpdateButtonAnimation();
+                }
+            }
+        }
+
+        protected virtual void Awake()
+        {
+            if (focusedButtonId == 0)
+            {
+                focusedButtonId = Animator.StringToHash("Focused");
+            }
+
+            if (selectedButtonId == 0)
+            {
+                selectedButtonId = Animator.StringToHash("Selected");
+            }
+
+            if (deHydrateButtonId == 0)
+            {
+                deHydrateButtonId = Animator.StringToHash("Dehydrate");
+            }
+
+            if (stayFocusedButtonId == 0)
+            {
+                stayFocusedButtonId = Animator.StringToHash("StayFocused");
+            }
+        }
+
+        protected virtual void OnEnable()
+        {
+
+            // Set the initial alpha
+            if (ToolTipRenderer != null)
+            {
+                Color tipColor = ToolTipRenderer.material.color;
+                tipColor.a = 0.0f;
+                ToolTipRenderer.material.color = tipColor;
+                toolTipTimer = 0.0f;
+            }
+
+            UpdateVisuals();
+            UpdateButtonAnimation();
+        }
+
+        private void Update()
+        {
+            if (ToolTipRenderer != null && (Focused && toolTipTimer < ToolTipFadeTime) || (!Focused && toolTipTimer > 0.0f))
+            {
+                // Calculate the new time delta
+                toolTipTimer = toolTipTimer + (Focused ? Time.deltaTime : -Time.deltaTime);
+
+                // Stop the timer if it exceeds the limit.  Clamp doesn't work here since time can be outside the normal range in some situations
+                if (Focused && toolTipTimer > ToolTipFadeTime)
+                {
+                    toolTipTimer = ToolTipFadeTime;
+                }
+                else if (!Focused && toolTipTimer < 0.0f)
+                {
+                    toolTipTimer = 0.0f;
+                }
+
+                // Update the new opacity
+                if (ToolTipRenderer != null)
+                {
+                    Color tipColor = ToolTipRenderer.material.color;
+                    tipColor.a = Mathf.Clamp(toolTipTimer, 0, ToolTipFadeTime) / ToolTipFadeTime;
+                    ToolTipRenderer.material.color = tipColor;
+                }
+            }
+        }
+
+        public void DehydrateButton()
+        {
+            if (ButtonAnimator != null && ButtonAnimator.isInitialized)
+            {
+                var animatorHashes = ButtonAnimator.parameters;
+                for (int i = 0; i < animatorHashes.Length; i++)
+                {
+                    if (animatorHashes[i].nameHash == deHydrateButtonId)
+                    {
+                        ButtonAnimator.SetTrigger(deHydrateButtonId);
```

Comment:
I think it's a good idea to let developers figure out how they want to handle buttons and just have an example.